### ISSUE TITLE
Midterm Refactoring and Bugfixing

### ISSUE THREAD

**Issue by MikeFAU** [amosproj/amos2024ss07-updating-flash-boot-loader#81]
## User story
1. As a software developer
2. I want to refactor and bugfix the current implementations of GUI and MCU
3. So that the SW for MCU and GUI is stable for the Midterm Release

## Acceptance criteria (To be completed until Midterm Release)
* GUI: Logging to UI Console is updated to necessary logs (CAN Driver, UDS Messages)
* MCU: UDS Refactoring is done
* GUI: Improvement of UDS message interpreter 
* Testing: Further testcases are added for communication between GUI and MCU

## Definition of done (DoD)
* Added only after week 5
* The same for all features
* Here goes the project specific part

## DoD general criteria
* Refactoring has been fully implemented
* Known Bugs have been solved 
* Implementation has been merged into the mainline
* All tests are passing
* Developers agreed to release
